src/vs/platform/actions/browser/menuEntryActionViewItem.ts

```diff
@@ -317,6 +318,10 @@ export class SubmenuEntryActionViewItem extends DropdownMenuActionViewItem {
 
 		container.classList.add('menu-entry');
 		const action = <SubmenuItemAction>this._action;
+		const toggled: ContextKeyExpression | undefined = action.item.toggled;
+		if (toggled && toggled.evaluate(this._contextKeyService.getContext(container))) {
```

Comment:
This shouldn't be here. The action is just displaying/representing its properties and shouldn't evaluate/recompute ones. There is the toggled/checked property on the action which could be used but TBH I have concerns (see below)